In [10]:
# Import all required modules
import pandas as pd
import numpy as np

# Disable warnings
import warnings
warnings.filterwarnings("ignore")

# Import plotting modules
import seaborn as sns
sns.set()
import matplotlib
import matplotlib.pyplot as plt
import matplotlib.ticker
%matplotlib inline

In [11]:
# Tune the visual settings for figures in `seaborn`
sns.set_context(
    "notebook", 
    font_scale=1.5,       
    rc={ 
        "figure.figsize": (11, 8), 
        "axes.titlesize": 18 
    }
)
from matplotlib import rcParams
rcParams['figure.figsize'] = 11, 8

In [12]:
data=df = pd.read_csv('../data/mlbootcamp5_train.csv', sep=';')
df.head()

id    age  gender  height  weight  ap_hi  ap_lo  cholesterol  gluc  smoke  \
0   0  18393       2     168    62.0    110     80            1     1      0   
1   1  20228       1     156    85.0    140     90            3     1      0   
2   2  18857       1     165    64.0    130     70            3     1      0   
3   3  17623       2     169    82.0    150    100            1     1      0   
4   4  17474       1     156    56.0    100     60            1     1      0   

   alco  active  cardio  
0     0       1       0  
1     0       1       1  
2     0       0       1  
3     0       1       1  
4     0       0       0

**Question 1.1. (1 point). How many men and women are present in this dataset?**

In [23]:
data.groupby('gender')['height'].mean()

gender
1    161.355612
2    169.947895
Name: height, dtype: float64

In [18]:
data['gender'].value_counts()

1    45530
2    24470
Name: gender, dtype: int64

**Answer(1.1):45530 women and 24470 men**

**Question 1.2. (1 point).Which gender more often reports consuming alcohol - men or women?**

In [26]:
data.groupby('gender')['alco'].value_counts(normalize=True)

gender  alco
1       0       0.974500
        1       0.025500
2       0       0.893625
        1       0.106375
Name: alco, dtype: float64

**Answer(1.2):men**

**Question 1.3. (1 point). What is the difference between the percentages of smokers among men and women (rounded)?**

In [51]:
data1 = data.groupby(['gender','smoke'])[['smoke']].count().rename(columns={'smoke': 'count'}).reset_index()

In [80]:
M = data1[(data1['gender']==2)&(data1['smoke']==1)]['count'].unique()[0]/data1[(data1['gender']==2)]['count'].sum()*100

In [81]:
F = data1[(data1['gender']==1)&(data1['smoke']==1)]['count'].unique()[0]/data1[(data1['gender']==1)]['count'].sum()*100

In [87]:
round(abs(F-M))

20.0

**Answer(1.3): 20**

**Question 1.4. (1 point). What is the difference between median values of age for smokers and non-smokers (in months, rounded)?**

In [75]:
data1[(data1['gender']==1)]['count'].sum()

45530

In [98]:
round(abs(data[data['smoke']==1]['age'].median()/30-data[data['smoke']==0]['age'].median()/30))

20.0

**Answer(1.4): 20**

**Question 1.5. (2 points). Calculate the fraction of the people with CVD for the two segments described above. What is the ratio of these two fractions?**

In [37]:
data['age_years'] = round(((data['gender']==2)&(data['age']<=23360)\
                            &(21900<=data['age']))*data['age']/365).astype('int64')

In [38]:
old_smoke = data[(data['age_years']>=60)&(data['smoke']==1)]

In [66]:
mini = old_smoke[(old_smoke['ap_hi']<120)&(old_smoke['cholesterol']==1)&(old_smoke['cardio']==1)].shape[0]

In [67]:
maxi = old_smoke[(old_smoke['ap_hi']>=160)&(old_smoke['ap_hi']<180)&(old_smoke['cholesterol']==3)&(old_smoke['cardio']==1)].shape[0]

In [68]:
round(mini/maxi)

1

**Answer(1.5): 1**